torch/csrc/jit/passes/onnx/remove_inplace_ops_for_onnx.cpp

```diff
@@ -632,29 +653,30 @@ void trackAndRegisterAttributesInBlocks(
       // If inside a block, keep the output value to register in block
       // output.
       auto block_ = n->owningBlock();
+      Node* cloneNode = insertCloneBeforeNode(graph, n->inputs().at(1), n);
       if (block_->owningNode() &&
-          block_->owningNode()->kind() == prim::If) { // TODO: Add loop
-
+          (block_->owningNode()->kind() == prim::If ||
+           block_->owningNode()->kind() == prim::Loop)) {
         auto attrValue = (setAttrValues.find(fullName) != setAttrValues.end())
             ? setAttrValues[fullName]
             : allAttrValues[fullName];
 
         auto blockOutput = registerSetAttrInBlocks(
-            graph, block_, n->inputs().at(1), attrValue, fullName);
+            graph, block_, cloneNode, attrValue, fullName);
+
         nextSetAttrValues[fullName] = blockOutput;
       }
       // SetAttr writes a value to an attr. Keep this
       // in the setAttrValues map.
-      setAttrValues[fullName] = n->inputs().at(1);
+      setAttrValues[fullName] = cloneNode->output();
     }
   } else if (n->kind() == prim::GetAttr) { // Handle GetAttr node
     if (setAttrValues.find(fullName) != setAttrValues.end()) {
       // Attr has been set earlier in the graph.
       // Read its value from setAttrValues map.
       auto set_attr_node_input = setAttrValues[fullName];
       // Clone SetAttr input
-      auto cloneNode = insertCloneBeforeNode(graph, set_attr_node_input, n);
-      n->output()->replaceAllUsesAfterNodeWith(n, cloneNode->output());
+      n->output()->replaceAllUsesAfterNodeWith(n, set_attr_node_input);
```

Comment:
no longer cloning setattr input?

Reply:
So, previously, any instance of setAttr was being recorded, and a clone was inserted in-place of a getAttr node on an attribute that had been set before.
So, if we had such a getAttr in a loop block, we could not trace the clone node input (coming from settAttr) as the loop inputs.
Now this issue is resolved.